src/control/actor/host/notify.go

```diff
@@ -60,14 +60,28 @@ func (c *HostNotifyCmd) connectedF(net network.Network, conn network.Conn) {
 
 	// request metadata as soon as we connect to a peer
 	peerData, err := PollPeerMetadata(conn.RemotePeer(), c.Base, c.PeerMetadataState, c.Store, c.PeerStore)
+	// request BeaconStatus metadata as we connect to a peer
+	h, err := c.Host()
+	if err != nil {
+		log.Error(err)
+	}
+	bState, err2 := nodemetadata.ReqBeaconStatus(context.Background(), h, conn.RemotePeer())
+	if err2 != nil {
+		log.Warn(err)
+	}
+	logrus.Info(&bState)
 	var peer metrics.Peer
+	// TODO: so far, If we didn't manage to exchange metadata, we asume that the peer didn't exchange Status neither
+	// 		  IMPORTANT: Metadata doen't include any kind of info related to the Host/Node
+	// 				     It just includes SeqNumber, Attnets
 	if err == nil {
-		peer = fetchPeerExtraInfo(peerData)
-		_ = peer
+		peer = fetchPeerExtraInfo(peerData, bState)	
```

Comment:
I would mix the beacon state with the metadata. We could add two more methods to the peer.
* UpdateMetadata
* UpdateBeaconStatus

Each funtion updates the metadata/beaconstatus of the peer and also creates a time stamp on when it was fetched.